### DISCUSSION TITLE
Set QTable columns to auto-stretch within table width but not beyond.

### DISCUSSION THREAD

**Discussion by jeff-simeon** [quasarframework/quasar#11864] [General - Components / Directives / etc] [answered]
I have a QTable that I want to stretch to 100% of the available width and height of the parent container but not beyond. I want to columns of the table to auto-stretch to fit content, but only within the constraints of the table.

Right now I have the problem that the table width expands too wide when the columns try to auto-size.

E.g. this looks good:
![image](https://user-images.githubusercontent.com/87453236/147885627-f97534f6-388e-40d9-a3ca-b5f812d46b87.png)

this doesn't
![image](https://user-images.githubusercontent.com/87453236/147885665-66d9337b-caa9-4640-9732-c07901067cd1.png)

I have the following QTable 
![image](https://user-images.githubusercontent.com/87453236/147885708-97057c2d-eacd-437f-b5b4-3d842a15b7f3.png)

**Comment by metalsadman** [comment, marked as answer]
I think you'll need to set some fix widths on some columns that you expect to have wider content, as you can see in your pic specifically in the `Baselines` column.
